### PR TITLE
[SPARK-48376][SQL][FOLLOWUP] Add ITERATE statement fixes

### PR DESCRIPTION
**Pull request by davidm-db**
### What changes were proposed in this pull request?
Previous [pull request](https://github.com/apache/spark/pull/47973) introduced new logic for `LEAVE` and `ITERATE` statements, but I missed to introduce minor part of `ITERATE` statement handling.
While `ITERATE` statement is not allowed for compounds (i.e. user cannot use `LEAVE <label>` where label belongs to compound) it's still possible to use `ITERATE` statement within the compound (i.e. when using label of the surrounding `WHILE` loop for example). In such cases, it's required to drop variables defined in the compound (the same way as for `ITERATE` statement).
We don't yet have execution logic (I will work on POC and design doc during the next week), but this is setting up stuff now, so we don't miss anything in future.

### Why are the changes needed?
Changes are minor, they improve consistency among test suites for SQL scripting.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Added small test, but not completely related to this change. Since we don't have execution logic yet, it's hard to test. I will add more tests together with the execution logic.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by davidm-db** [comment]
@cloud-fan @MaxGekk

File: sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala [hunk 1]
```diff
@@ -678,4 +678,38 @@ class SqlScriptingInterpreterSuite extends QueryTest with SharedSparkSession {
...
+        |END""".stripMargin
+    val expected = Seq(
+      Seq.empty[Row], // declare
+      Seq.empty[Row], // set x = 0
+      Seq.empty[Row], // set x = 1
+      Seq(Row(1)), // select 1
+      Seq(Row(2)), // select 1
+      Seq.empty[Row], // set x= 2
+      Seq(Row(1)), // select 1
+      Seq(Row(2)), // select 1
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala:709]:
```suggestion
      Seq(Row(2)), // select 2
```

File: sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala [hunk 2]
```diff
@@ -678,4 +678,38 @@ class SqlScriptingInterpreterSuite extends QueryTest with SharedSparkSession {
...
+        |  END WHILE;
+        |  SELECT x;
+        |END""".stripMargin
+    val expected = Seq(
+      Seq.empty[Row], // declare
+      Seq.empty[Row], // set x = 0
+      Seq.empty[Row], // set x = 1
+      Seq(Row(1)), // select 1
+      Seq(Row(2)), // select 1
+      Seq.empty[Row], // set x= 2
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala:707]: nit:
```suggestion
      Seq.empty[Row], // set x = 2
```

File: sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala [hunk 3]
```diff
@@ -206,18 +213,36 @@ abstract class CompoundNestedStatementIteratorExec(
...
+      // TODO: Add interpreter tests as well.
 
       // Check if label has been matched.
       leaveStatement.hasBeenMatched = label.isDefined && label.get.equals(leaveStatement.label)
     }
   }
+
+  /**
+   * Actions to do when ITERATE statement is encountered,
+   *   to stop the execution of this compound.
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala:232]: nit:
```suggestion
   * Actions to do when ITERATE statement is encountered, to stop the execution of this compound.
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala (hunk 3 above)
Comment (davidm-db) [reply sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala]: Done.

File: sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala (hunk 2 above)
Comment (davidm-db) [reply sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala]: Done.

File: sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala (hunk 1 above)
Comment (davidm-db) [reply sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala]: Done.

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @davidm-db.

**Pull request closed without merging**